adapter/thing_management.md

```diff
@@ -0,0 +1,404 @@
+    * Client sends `cmd.thing.pin` command with the PIN to the adapter.
+* Once inclusion and interview of the device is finished adapter generates `evt.thing.inclusion_report` event with the thing specification.
+* Client sends `cmd.thing.inclusion` command to the adapter with `false` value to stop the inclusion process. Regardless, adapter should cancel inclusion after a specified timeout.
+
+### Smart Start Inclusion
+
+Smart Start inclusion is a Z-Wave specific flow allowing to preliminarily add a device to a provisioning list so that it can be automatically included once the device is turned on.
+See [Smart Start](/adapter/zwave_smart_start.md) documentation for more details.
+
+### Custom Inclusion
+
+Adapters utilizing custom protocol communication through local network or third-party cloud will have their own specific inclusion flow specified it in their own documentation.
+Preliminary steps of such flows will often include authentication with a third-party system, specific configuration, and selecting devices to be connected to the system.
+At the end of this process the adapter is always required to generate `evt.thing.inclusion_report` event with the thing specification.
+
+### Network Exclusion
+
+Network exclusion flow is used by wireless network adapters to exclude devices from the network and remove them from the system.
+
+* Client sends `cmd.thing.exclusion` command to the adapter to start exclusion process and switch adapter into unpairing mode.
+* User interacts with a physical device to switch it into unpairing mode, e.g. clicks a button on a device three times in a quick sequence.
+* At any time the adapter can generate `evt.thing.exclusion_status_report` event to report the current status of the exclusion process.
+* Once exclusion is finished adapter generates `evt.thing.exclusion_report` event with the excluded thing address.
+
+### Forced Exclusion
+
+If a device is offline and unreachable or the adapter does not support the network exclusion flow, the device can be force deleted from the system using this flow.
+
+* Client sends `cmd.thing.delete` command to the adapter to force delete a thing.
+* Adapter removes the thing from the network and generates `evt.thing.exclusion_report` event with the excluded thing address.
+
+### Requesting Inclusion Report
+
+* At any time a client can request inclusion report by sending `cmd.thing.get_inclusion_report` command to the adapter with a specified address of an already paired thing.
+* Adapter generates `evt.thing.inclusion_report` event with the thing specification.
+
+## FIMP Specification
+
+### Topics
+
+Every adapter should have unique resource name and use the following topics in accordance with [topic format](/fimp/topic_format.md):
+
+`pt:j1/mt:cmd/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+`pt:j1/mt:evt/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+### Service name
+
+Every adapter should define its own service name.
+
+### Interfaces
```

Comment:
Zwave adapter also implements:
1. cmd.thing.reload_template - which is used by Thingsplex
2. cmd.thing.force_remove - I've asked other developers and noone is using it
3. cmd.thing.replace - I don't know whether it's used by any of services but it might be pretty usable for end user. Case:
    a. User added device X
    b. User has added automations for that device in our FH system 
    c. Device X has one broken functionality (e.g. Firmware with a bug)
    d. User wants to buy a newer version of that device but wants to keep automations and not create them once again.
    e. User uses "replace node" functionality for the broken device and includes the new one "in-place" of the old one.
Additionally this method not implemented "well enough" in zwave ad yet as it do not handle Security callbacks (but can easily be implemented).

4. cmd.network.reset - it's used when factory resetting a hub. But also should be available from SDU app in some advanced zwave section. It's described in adapter_service.md - shouldnt it be also described here?
5. cmd.network.update - same description as above
6. cmd.network.node_update - same description as aboce
7. cmd.adapter.get_states - I can see it's used in Thingsplex

Reply:
1. Looks like Z-Wave only thing and a semi-private API.
2. You can keep it as private alias for cmd.thing.delete or remove.
3. If it's not used by the app it's private API of the adapter.
4. Network based interfaces are in the not-refactored document.
5. Vide 4.
6. Vide 4.
7. Will look into it in next PR.